lib/compiler.js

```diff
@@ -54,28 +56,26 @@ function getDeps(requires) {
       }
 
       if (requires[lib].ym) {
-        deps.ym.push(requires[lib].ym);
-        deps.ymVars.push(lib);
-        deps.ymLibs.push(lib);
+        deps.ymNames.push(lib);
+        deps.ym[lib] = requires[lib].ym;
       }
+
+      deps.ymLibs.push(lib);
     }
   }
 
   if (deps.ymLibs.length) {
-    deps.ymLibs = 'engine.libs = {' + deps.ymLibs.map(function(item) {
-      return '"' + item + '":' + item;
-    }).join() + '};';
-  }
-
-  if (deps.ymVars.length) {
-    deps.ymVarsString = deps.ymVars.map(function(item) {
-      return item + ':' + item;
-    }).join('');
+    deps.ymVars = ',' + deps.ymLibs.map(function (item) {
```

Comment:
we make 3 maps here, can do 1 for instead, could be faster. what do you think?

Reply:
@dustyo-O `generate()` is not runtime operation. No matter how it fast or slow we use it to generate bundles for browser.